SimpleSDMPolygons/src/operations/clip.jl

```diff
@@ -0,0 +1,34 @@
+function _get_polygon_from_bbox(bbox::NamedTuple)
+    all(haskey(bbox, k) for k in [:left, :bottom, :right, :top]) || throw(ArgumentError("Bounding box tuple doesn't have correct keys. It must contain :top, :bottom, :left, and :right"))
+
+    bl, tl, tr, br = (bbox[:left], bbox[:bottom]),
+        (bbox[:left], bbox[:top]),
+        (bbox[:right], bbox[:top]),
+        (bbox[:right], bbox[:bottom])
+    return Polygon(bl, tl, tr, br)
+end 
+
+
+function clip(
+    feat::Feature,
+    bbox::NamedTuple
+)
+    bbox_poly = _get_polygon_from_bbox(bbox)
+    return Feature(intersect(feat, bbox_poly), feat.properties)
+end
+
+function clip(
+    fc::FeatureCollection,
+    bbox::NamedTuple
+)
+    bbox_poly = _get_polygon_from_bbox(bbox)
+    return intersect(add(fc), bbox_poly)
```

Comment:
i think this is the line - shouldn't this be vcat? or a third argument to pick add or vcat?

Reply:
yeah `intersect` and `subtract` both were aggregating `FeatureCollection`s before applying the operation, which typically wouldn't make sense because you'd want to preserve the distinct features. `add` still aggregates FCs prior to combining because vcat is an alternative, i should add a bit in the docs to reflect this

Reply:
add also has an additional method that only works on FCs, so this result could be passed to `add` to get a single aggregated polygon

Reply:
yeah i think adding a quick mention of this in the doc would be good - i can see cases where either behavior is required